okio/src/main/java/okio/RealBufferedSource.java

```diff
@@ -233,6 +233,16 @@ public RealBufferedSource(Source source) {
     return buffer.readLongLe();
   }
 
+  @Override public long readDecimalLong() throws IOException {
+    request(20);
```

Comment:
request != require, so it would take the 5 chars and then move on to parsing and succeed. I was worried about the other case where you have 5 but there's data on the wire that hasn't come across yet. In general, it's clear this needs to go byte-by-byte with request(1) looking for matching chars and then parse later, like you said.